packages/Meshfree/src/DTK_DetailsSVDImpl.hpp

```diff
@@ -0,0 +1,304 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#ifndef DTK_DETAILS_SVD_IMPL_HPP
+#define DTK_DETAILS_SVD_IMPL_HPP
+
+#include <Kokkos_Core.hpp>
+
+#include <Kokkos_ArithTraits.hpp>
+
+namespace DataTransferKit
+{
+namespace Details
+{
+
+template <typename DeviceType>
+struct SVDFunctor
+{
+  public:
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    using matrices_type = Kokkos::View<double *, DeviceType>;
```

Comment:
I did talk with @aprokop about this - this is related to possibly arbitrary sized matrices being operated on in batches

Reply:
Another reason to store them flat is because we think this may be what a batched interface is going to look like. It is the most generic way to store them.